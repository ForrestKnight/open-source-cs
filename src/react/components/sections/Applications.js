/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Applications extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Applications" className="Applications">
                
            </div>
        );
    }
}


export default Applications;