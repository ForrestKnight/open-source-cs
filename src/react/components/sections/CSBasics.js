/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class CSBasics extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="CSBasics" className="CSBasics">
                
            </div>
        );
    }
}


export default CSBasics;