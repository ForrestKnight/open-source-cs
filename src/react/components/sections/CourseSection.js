/* Packages */
import React from 'react';
import Course from '../Course';


/* Component */
export class CourseSection extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id={this.props.name} className="CourseSection">
                
            </div>
        );
    }
}


export default Math;