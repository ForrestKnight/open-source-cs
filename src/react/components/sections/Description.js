/* Packages */
import React from 'react';


/* Component */
export class Description extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <section id="Description" className="Description">
                <header className="Description__header">
                    <h1 className="Description__title">
                        The Open Source Computer Science Degree
                    </h1>
                </header>

                <section className="Description__description">
                    <p>
                        This is a curated list of free courses from reputable universities like MIT, Stanford, and Princeton that satisfy the same requirements as an undergraduate Computer Science degree, minus general education.
                    </p>

                    {/* Embed video, different sizes for different screens*/}
                    <figure className="Description__videoPhone">
                        <iframe width="300" height="168" src="https://www.youtube.com/embed/NyOvFSP_IpQ" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
                        
                        <figcaption>
                            YouTuber <a href="https://www.youtube.com/channel/UC2WHjPDvbE6O328n17ZGcfg">Forrest Knight</a> explains the concept of Open Source CS
                        </figcaption>
                    </figure>

                    <figure className="Description__videoTablet">
                        <iframe width="560" height="315" src="https://www.youtube.com/embed/NyOvFSP_IpQ" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

                        <figcaption>
                            YouTuber <a href="https://www.youtube.com/channel/UC2WHjPDvbE6O328n17ZGcfg" target="_blank" rel="noopener">Forrest Knight</a> explains the concept of Open Source CS
                        </figcaption>
                    </figure>

                    <p>
                        The <a href="https://github.com/ForrestKnight/open-source-cs" target="_blank" rel="noopener">original Open Source CS repository</a> is just a table in a README file on GitHub. This website gives a more accessible and responsive alternative. 
                    </p>
                </section>
            </section>
        );
    }
}


export default Description;