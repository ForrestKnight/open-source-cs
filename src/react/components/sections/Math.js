/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Math extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Math" className="Math">
                
            </div>
        );
    }
}


export default Math;