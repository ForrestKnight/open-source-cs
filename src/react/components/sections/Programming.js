/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Programming extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Programming" className="Programming">
                
            </div>
        );
    }
}


export default Programming;