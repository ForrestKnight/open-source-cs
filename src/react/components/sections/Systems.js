/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Systems extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Systems" className="Systems">
                
            </div>
        );
    }
}


export default Systems;