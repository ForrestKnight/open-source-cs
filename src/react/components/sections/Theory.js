/* Packages */
import React from 'react';
import {connect} from 'react-redux';


/* Component */
export class Theory extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div id="Theory" className="Theory">
                
            </div>
        );
    }
}


export default Theory;