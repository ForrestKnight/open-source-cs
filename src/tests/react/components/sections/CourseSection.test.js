/* Packages */
import React from 'react';
import {shallow} from 'enzyme';


/* Component */
import {CourseSection} from '../../../../react/components/sections/CourseSection';


/* Test Config */
let component;
beforeEach(() => {
    component = shallow(<CourseSection />);
});


/* Tests */
test('Render CourseSection section', () => {
    expect(component).toMatchSnapshot();
});

