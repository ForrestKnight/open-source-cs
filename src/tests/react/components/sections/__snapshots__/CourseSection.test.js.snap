// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render CourseSection with courses 1`] = `
<article
  className="CourseSection"
  id="SectionName"
>
  <header
    className="CourseSection__header"
  >
    <h2>
      Section Name
    </h2>
  </header>
  <section
    className="CourseSection__courses"
  >
    <Course
      key="Course 1"
      name={
        Object {
          "text": "Course 1",
        }
      }
    />
    <Course
      key="Course 2"
      name={
        Object {
          "text": "Course 2",
        }
      }
    />
  </section>
</article>
`;

exports[`Render CourseSection without courses 1`] = `
<article
  className="CourseSection"
  id="SectionName"
>
  <header
    className="CourseSection__header"
  >
    <h2>
      Section Name
    </h2>
  </header>
  <section
    className="CourseSection__courses"
  />
</article>
`;
